lib/dry/system/booter.rb

```diff
@@ -115,12 +131,21 @@ def call(name_or_component)
 
       # @api private
       def boot_dependency(component)
-        boot_file = boot_file(component)
+        name = component.respond_to?(:root_key) ? component.root_key : component.to_sym
 
-        start(boot_file.basename(".*").to_s.to_sym) if boot_file
+        if (component = find_component(name))
+          start(component)
+        end
       end
 
-      # @api private
+      # Returns all boot files within the configured paths
+      #
+      # Searches for files in the order of the configured paths. In the case of multiple
+      # identically-named boot files within different paths, the file found first will be
+      # returned, and other matching files will be discarded.
+      #
+      # @return [Array<Pathname>]
+      # @api public
```

Comment:
Added this comment as a clear reminder that we rely on this from other gems, namely dry-rails, effectively making it public API.